### ISSUE TITLE
False positive for Style/UnneededCondition

### ISSUE THREAD

**Issue by dmarcoux** [rubocop-hq/rubocop#7340]
I work on a _Ruby on Rails_ project in which I believe we stumbled on a false positive for the `Style/UnneededCondition` cop. I added a code example below to reproduce the false positive. It does contain _Ruby on Rails_ specific code, but I kept it to the minimum (only kept `render`).

## Steps to reproduce the problem

```ruby
def some_rails_action
  package = MyPackageModel.do_something
  if package
    target_project = package.project.name
    target_package = package.name
    description = package.commit_message(target_project, target_package)
  end

  render json: {
    'targetProject': defined? target_project ? target_project : '',
    'targetPackage': defined? target_package ? target_package : '',
    'description': defined? description ? description : '',
    'somethingElse': '12345'
  }
end
```

I wonder if there's better way to write this code. Repeating the `if package` in `render json` isn't a viable solution. Setting the local variables (`target_project` and the others) beforehand could also work.

## Actual behavior

The example above will detect an offense:
```
C: Style/UnneededCondition: Use double pipes || instead.
      'targetProject': defined? target_project ? target_project : '',
```

## Expected behavior

I expected Rubocop to not detect an offense.

## RuboCop version

```
0.74.0 (using Parser 2.6.4.0, running on ruby 2.5.5 x86_64-linux-gnu)
```

**Comment by koic** [comment]
This does not seem to be a false positive.
Is your expectation the following code?

```ruby
defined?(target_project) ? target_project : ''
```

In that case, parentheses cannot be omitted. Omitting parentheses means the following:

```ruby
defined?(target_project ? target_project : '')
```

Examples for each behavior.

```console
% ruby -e "foo = 'a'; p defined? foo ? true : false"
"expression"
% ruby -e "foo = 'a'; p defined?(foo) ? true : false"
true
% ruby -e "foo = 'a'; p defined?(foo ? true : false)"
"expression"
```

**Comment by dmarcoux** [comment]
Thank you @koic for your reply. I indeed expect the first code block that you wrote in your comment (so `defined?(target_project) ? target_project : ''`). I didn't realize in this case that parentheses cannot be omitted.

You are right that it's not a false positive. It's an error on my side, so I'm closing this issue. Thank you for RuboCop (same goes to everybody else involved in the project)
